ds2482.c

```diff
@@ -0,0 +1,461 @@
+/*  
+    sensord - Sensor Interface for XCSoar Glide Computer - http://www.openvario.org/
+    Copyright (C) 2014  The openvario project
+    A detailed list of copyright holders can be found in the file "AUTHORS" 
+
+    This program is free software; you can redistribute it and/or 
+    modify it under the terms of the GNU General Public License 
+    as published by the Free Software Foundation; either version 3
+    of the License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, see <http://www.gnu.org/licenses/>.	
+*/
+
+#include "ds2482.h"
+#include <time.h>
+#include <stdio.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+#include <math.h>
+#include <time.h>
+#include <linux/i2c-dev.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <string.h>
+#include "def.h"
+
+extern int g_debug;
+extern FILE *fp_console;
+
+//http://datasheets.maximintegrated.com/en/ds/DS2482-100.pdf
+//http://datasheets.maximintegrated.com/en/ds/DS18B20.pdf
+//http://www.wulfden.org/downloads/datasheets/DS2482_AN3684.pdf
+
+// The routines in this code were roughly lifted from https://pastebin.com/0d93ZuRb
+// Those assume the use of the Search and Match ROM.  As such a number of
+// functions are included that are not needed if a Skip ROM is used which
+// is adequate, and faster if only a single 1 Wire device is present.
+
+int ds2482_open(t_ds2482 *sensor, unsigned char i2c_address)
+{
+	// local variables
+	int fd;
+	
+	// try to open I2C Bus
+	fd = open("/dev/i2c-1", O_RDWR);
+	
+	if (fd < 0) {
+	   fprintf(stderr, "Error opening file: %s\n", strerror(errno));
+	   return 1;

```

Comment:
Please don't mix spaces and tabs. This project seems to use tabs as indentation characters. Let's keep it consistent.